services/ymax-planner/src/engine.ts

```diff
@@ -192,6 +193,57 @@ type ProcessPortfolioPowers = Pick<
   };
 };
 
+/**
+ * Determines if a plan needs to be submitted for a given portfolio flow.
+ * A plan is needed if:
+ * - The flow does not have vstorage data (flowKeys does not contain flowKey), OR
+ * - The flow exists but has status: 'init'
+ *
+ * @param portfolioKey - The portfolio key (e.g., "portfolio1")
+ * @param flowKey - The flow key (e.g., "flow1")
+ * @param flowKeys - Set of flow keys that have vstorage data
+ * @param vstorage - Vstorage query interface
```

Comment:
have the caller do the vstorage IO. pass in the `status` or, in case there is no node, `undefined`.